src/validators/arguments_v3.rs

```diff
@@ -0,0 +1,654 @@
+                        }
+                    } else {
+                        // Required and no default, error:
+                        match parameter.mode {
+                            ParameterMode::PositionalOnly => {
+                                errors.push(ValLineError::new_with_loc(
+                                    ErrorTypeDefaults::MissingPositionalOnlyArgument,
+                                    original_input,
+                                    index,
+                                ));
+                            }
+                            ParameterMode::PositionalOrKeyword => {
+                                errors.push(lookup_key.error(
+                                    ErrorTypeDefaults::MissingArgument,
+                                    original_input,
+                                    self.loc_by_alias,
+                                    &parameter.name,
+                                ));
+                            }
+                            ParameterMode::KeywordOnly => {
+                                errors.push(lookup_key.error(
+                                    ErrorTypeDefaults::MissingKeywordOnlyArgument,
+                                    original_input,
+                                    self.loc_by_alias,
+                                    &parameter.name,
+                                ));
+                            }
+                            _ => unreachable!(),
+                        }
+                    }
+                }
+            }
+        }
+
+        // if there are args check any where index > positional_params_count since they won't have been checked yet
+        if let Some(args) = args_kwargs.args() {
+            let len = args.len();
+            if len > self.positional_params_count {
+                if let Some(var_args_param) = self.parameters.iter().find(|p| p.mode == ParameterMode::VarArgs) {
+                    for (index, item) in args.iter().enumerate().skip(self.positional_params_count) {
+                        match var_args_param.validator.validate(py, item.borrow_input(), state) {
+                            Ok(value) => output_args.push(value),
+                            Err(ValError::LineErrors(line_errors)) => {
+                                errors.extend(line_errors.into_iter().map(|err| err.with_outer_location(index)));
+                            }
+                            Err(err) => return Err(err),
+                        }
+                    }
+                } else {
+                    for (index, item) in args.iter().enumerate().skip(self.positional_params_count) {
+                        errors.push(ValLineError::new_with_loc(
+                            ErrorTypeDefaults::UnexpectedPositionalArgument,
+                            item,
+                            index,
+                        ));
+                    }
```

Comment:
I think extra handling may be relevant here?

Reply:
This would align with the extra behavior for kwargs, but the current arguments schema doesn't apply anything. I'll leave it as is for now and mention in it in the follow up issue.